qcodes/instrument_drivers/tektronix/Keithley_3706A.py

```diff
@@ -0,0 +1,742 @@
+        A helper function that gets two channel names from the available
+        channels list and join them via a colon to define a channel range.
+        """
+        range_list = []
+        for i in self._get_slot_ids():
+            channel = self.get_channels_by_slot(int(i))
+            for element in itertools.combinations(channel, 2):
+                range_list.append(':'.join(element))
+        return range_list
+
+    def get_channels(self) -> List[str]:
+        """
+        This function returns the name of the matrix channels.
+        User can call this function to see the names of the available
+        channels, in case he/she is not familiar with the naming convention.
+        However, note that, this is a standalone helper function and
+        the usage of channel attributes of the instrument driver does
+        not depend on the functionality of this method.
+        """
+        slot_id = self._get_slot_ids()
+        row_list = self._get_rows()
+        column_list = self._get_columns()
+        matrix_channels = []
+        for i, slot in enumerate(slot_id):
+            for element in itertools.product(slot, row_list[i], column_list[i]):
+                matrix_channels.append(''.join(element))
+        return matrix_channels
+
+    def get_channels_by_slot(self, slot_no: int) -> List[str]:
+        """
+        Returns the channel names of a given slot.
+
+        Args:
+            slot_no: An integer value specifying the slot number.
+        """
+        slot_id = self._get_slot_ids()
+        if str(slot_no) not in slot_id:
+            raise UnknownOrEmptySlot("Please provide a valid slot identifier. "
+                                     f'Available slots are {slot_id}.')
+        row_list = self._get_rows()
+        column_list = self._get_columns()
+        matrix_channels_by_slot = []
+        for element in itertools.product(str(slot_no), row_list[0],
+                                         column_list[0]):
+            matrix_channels_by_slot.append(''.join(element))
+        return matrix_channels_by_slot
+
+    def get_analog_backplane_specifiers(self) -> List[str]:
+        """
+        Returns a list of comma separated strings representing available analog
+        backplane relays. This function should not be mixed with the
+        `get_backplane` method. The latter returns backplane relays which are
+        associated with a channel by using `set_backplane` method.
+        """
+        backplane_common_number = '9'
+        backplane_relay_common_numbers = ['11', '12', '13', '14', '15', '16']
+        slot_id = self._get_slot_ids()
+        analog_backplane_relays = []
+        for element in itertools.product(slot_id, backplane_common_number,
+                                         backplane_relay_common_numbers):
+            analog_backplane_relays.append(''.join(element))
+        return analog_backplane_relays
+
+    def connect_or_disconnect_row_to_columns(self, action: str,
```

Comment:
could you explain the choice for the API being "one method for connect/disconnect actions with first argument being the flag"? as opposed to, for example, "two separate methods, one for connect, other for disconnect" (removing the need for the flag argument and making the method names shorter and more direct).

Reply:
I already explained above, while responding to the relevant query, that creating a separate method will be a code repetition. If we assume for a moment that we have two functions for one to connect and for other to disconnect, these function will differ, effectively, only with a single line, i.e., execution of connect/disconnect. Thus, the latter implementation, IMHO, would decrease the code quality. With the current implementation, we have the desired functionality without duplication. I also added the usage of these functions to the example notebook, so users should be able to benefit from them in no time.

Reply:
I suggest that we make 2 individual methods that both call this one with the appropriate flat set and then make this method private. That will not give code repetition and create a cleaner api